api/clients/v2/accountant.go

```diff
@@ -87,18 +88,19 @@ func (a *Accountant) BlobPaymentInfo(ctx context.Context, numSymbols uint64, quo
 
 	overflowBinRecord := a.GetRelativeBinRecord(currentReservationPeriod + 2)
 	// Allow one overflow when the overflow bin is empty, the current usage and new length are both less than the limit
-	if overflowBinRecord.Usage == 0 && relativeBinRecord.Usage-numSymbols < binLimit && numSymbols <= binLimit {
-		overflowBinRecord.Usage += relativeBinRecord.Usage - binLimit
+	overflowBinRecord.Usage += relativeBinRecord.Usage - binLimit
+	if overflowBinRecord.Usage == relativeBinRecord.Usage-binLimit && relativeBinRecord.Usage-symbolUsage < binLimit && symbolUsage <= binLimit {
```

Comment:
not sure if i understand this change, it seems like this condition is same as before but we preemptively add the overflow usage before the check. 
why was this change necessary?

Reply:
Yep, we added this so that we will always add to the bin if reservation is used, otherwise both current and overflow bin gets reverted when the accountant uses ondemand (rollback added for overflow at [line](https://github.com/Layr-Labs/eigenda/pull/1034/files/c75419c7617f9314434bb69f7920e9091823dc6b#diff-12b74ec71c12c2e17777a7dee108651d3a1cd184a32a2bb522a43c0f0baa204dR101))